### PR TITLE
Add type hints according to PEP 484 and PEP 604

### PR DESCRIPTION
**Pull request by TonyCrane**
Add type hints according to [PEP 484](https://www.python.org/dev/peps/pep-0484/), [PEP 526](https://www.python.org/dev/peps/pep-0526/) and [PEP 604](https://www.python.org/dev/peps/pep-0604/)

## Motivation

The motivation for this is to allow the IDE (language server) to better analyze the code and give type hints like:
image
And this also provides convenience for writing API documentation in the next step

## Changes

I manually wrote type annotations for almost all class methods and functions following these PEP standards. 
However, due to the dynamic addition of attributes using `CONFIG` and `digest_config`, the types of some attribute values ​​cannot be annotated and therefore the whole module cannot be checked with static type checkers such as `mypy`. So this annotation is **exactly** like a **comment**.

And it's worth mentioning that in this pull request, I upgraded the python version available for ManimGL to python 3.7 and above (i.e. python 3.6 is no longer supported). Because the type annotation system has been relatively complete since 3.7, and numpy no longer supports 3.6 since 1.20 (the `numpy.typing` submodule has only been added since version 1.20)
I'm not sure if it's worth giving up on 3.6, but maybe that's a general trend.

Because the changes in this PR are almost type annotations, except for some additions of import, it will not affect the normal usage (python will not check whether the variable type satisfies the annotation at runtime)

By the way, in this pull request, I also changed the order of imports according to the PEP 8 (that is, `from __future__ import` first, then an empty line for built-in modules, an empty line for third-party modules, and an empty line for local modules)

## Progress
- [x] `manimlib.utils`
- [x] `manimlib.camera`
- [x] `manimlib.mobject`
    - [x]  `manimlib.mobject.mobject`
    - [x]  `manimlib.mobject.types`
    - [x]  `manimlib.mobject.svg`
- [x] `manimlib.animation`
- [x] `manimlib.event_handler`
- [x] `manimlib.scene`
- [x] `manimlib.shader_wrapper`
- [x] `manimlib.window`

### CODE REVIEW

File: manimlib/mobject/mobject.py [hunk 1]
```diff
@@ -86,23 +97,23 @@ def __init__(self, **kwargs):
     def __str__(self):
         return self.__class__.__name__
 
-    def __add__(self, other: 'Mobject') -> 'Mobject':
+    def __add__(self, other: "Mobject") -> "Mobject":
```
[the remark this answers is not in the document]
Comment (TonyCrane) [reply manimlib/mobject/mobject.py:100]: But only python 3.7+ can use that method.

[the remark this answers is not in the document]
Comment (TurkeyBilly) [reply manimlib/mobject/mobject.py:100]: > But only python 3.7+ can use that method.
Oh, sorry. Never mind then.

[the remark this answers is not in the document]
Comment (TonyCrane) [reply manimlib/mobject/mobject.py]: They are removed now. Because I decided to support only python 3.7+

**Comment by TonyCrane** [comment]
Wait for #1745

File: manimlib/mobject/svg/mtex_mobject.py [hunk 2]
```diff
@@ -431,12 +448,12 @@ def __init__(self, tex_string, **kwargs):
         self.scale(SCALE_FACTOR_PER_FONT_POINT * self.font_size)
 
     @staticmethod
-    def color_to_label(color):
+    def color_to_label(color: ManimColor) -> list:
```
Comment (TonyCrane) [inline manimlib/mobject/svg/mtex_mobject.py:451]: the return value is `int`

**Comment by TonyCrane** [comment]
I think this pull request needs more time to check, because it changes too much.
So any thoughts now? @3b1b

**Comment by 3b1b** [comment]
This is definitely a welcome push in the right direction. I can block out some time a bit later this week to give it a more thorough look.

You mention issues associated with the current "digest_config" usage. At some point, it would be worth getting rid of that practice entirely, and to instead put all the variables current in a CONFIG dict into the __init__ args, like normal python programmers :).

**Comment by YishiMichael** [comment]
I'd recommend that methods implemented in base classes and then overridden should keep input and output types maintained. As an example, some methods concerned with Mobject don't return self, while they do in an inherited class, which should be unified.

File: manimlib/mobject/svg/svg_mobject.py [hunk 3]
```diff
@@ -190,7 +194,10 @@ def get_mobject_from(self, shape):
         return None
 
     @staticmethod
-    def apply_style_to_mobject(mob, shape):
+    def apply_style_to_mobject(
+        mob: VMobject,
+        shape: se.Shape | se.Text
```
Comment (YishiMichael) [inline manimlib/mobject/svg/svg_mobject.py:199]: `shape` can be simply annotated with `se.GraphicObject`. Similar for `get_mobject_from` method.

File: manimlib/mobject/svg/svg_mobject.py [hunk 4]
```diff
@@ -200,16 +207,16 @@ def apply_style_to_mobject(mob, shape):
...
 
-    def line_to_mobject(self, line):
+    def line_to_mobject(self, line: se.Line) -> Line:
         return Line(
             start=_convert_point_to_3d(line.x1, line.y1),
             end=_convert_point_to_3d(line.x2, line.y2)
         )
 
-    def rect_to_mobject(self, rect):
+    def rect_to_mobject(self, rect: se.Rect) -> Rectangle | RoundedRectangle:
```
Comment (YishiMichael) [inline manimlib/mobject/svg/svg_mobject.py:219]: `RoundedRectangle` is inherited from `Rectangle`, so no need for multiple types here.

**Pull request merged**